x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -109,17 +114,49 @@ export const createAgentPolicyHandler: FleetRequestHandler<
   const esClient = context.core.elasticsearch.client.asInternalUser;
   const user = (await appContextService.getSecurity()?.authc.getCurrentUser(request)) || undefined;
   const withSysMonitoring = request.query.sys_monitoring ?? false;
+  const isDefaultFleetServer = request.body.is_default_fleet_server ?? false;
   const spaceId = context.fleet.spaceId;
+  // TODO set first default policy to is_default
+  // TODO installing these packages might take long (about 2 mins) which causes network timeout. this results in an error, and fails next agent policy creation request. Could this be improved by bundled default packages?
   try {
+    if (isDefaultFleetServer) {
+      // install fleet server package if not yet installed
+      await ensureInstalledPackage({
+        savedObjectsClient: soClient,
+        pkgName: FLEET_SERVER_PACKAGE,
+        esClient,
+      });
+    }
+    if (withSysMonitoring) {
+      // install system package if not yet installed
+      await ensureInstalledPackage({
+        savedObjectsClient: soClient,
+        pkgName: FLEET_SYSTEM_PACKAGE,
+        esClient,
+      });
+    }
+    if (request.body.monitoring_enabled?.length) {
+      // install elastic agent package if not yet installed
+      ensureInstalledPackage({
+        savedObjectsClient: soClient,
+        pkgName: FLEET_ELASTIC_AGENT_PACKAGE,
+        esClient,
+      });
+    }
     // eslint-disable-next-line prefer-const
     let [agentPolicy, newSysPackagePolicy] = await Promise.all([
       agentPolicyService.create(soClient, esClient, request.body, {
         user,
       }),
-      // If needed, retrieve System package information and build a new package policy for the system package
-      // NOTE: we ignore failures in attempting to create package policy, since agent policy might have been created
-      // successfully
-      withSysMonitoring
+      // TODO case when both isDefaultFleetServer and withSysMonitoring is true
```

Comment:
@joshdover is it a valid use case to add system monitoring to default fleet server policy? currently default fleet server policy only includes fleet server integration

Reply:
I think that is probably a valid use case for preconfiguration / kibana.yml, but isn't possible via the UI today.

Is there a major hurdle in supporting this case?

Reply:
I thought we are removing the default fleet server policy? I didn't read the code, so am I missing context behind this question?

Reply:
After reading the comments here and in design doc, it seems that we can completely get rid of the flag to mark a policy default (`is_default` and `is_default_fleet_server` flags in code). I wasn't aware of this before, probably because on the designs the policies were still called "Default policy" and "Default Fleet Server policy".

@joshdover I think we can support system integration with fleet server policy, I didn't want to start implementing it until confirmed.

Reply:
Makes sense to me. If we're going to drop the concept of default completely, we should remove that field from our Saved Object mappings and add a Saved Object migration that deletes this field.